blog/views.py

```diff
@@ -9,6 +10,40 @@
 from comment.forms import CommentForm
 from comment.models import Comments
 
+def is_post_owner(func):
+    def check(request, *args, **kwargs):
+        post_id = kwargs["post_id"]
+        post = Post.objects.get(id=post_id)
+        if post.user_id != request.user.id:
+            return HttpResponse("It is not yours ! You are not permitted !",
+                                status=403)
+        return func(request, *args, **kwargs)
+
+    return check
+
+def is_category_owner(func):
+    def check(request, *args, **kwargs):
+        category_id = kwargs["category_id"]
+        category = Category.objects.get(id=category_id)
+        if category.user_id != request.user.id:
+            return HttpResponse("It is not yours ! You are not permitted !",
+                                status=403)
+        return func(request, *args, **kwargs)
+
+    return check
+
+def is_user_owner(func):
+    def check(request, *args, **kwargs):
+        user_id = kwargs["user_id"]
```

Comment:
please leave these decorators in decorators.py located in the same directory of this view.py.